mantidproject/mantid
# Use std::array when possible

Description of work.

**To test:**

Code review, all tests should pass. I reviewed:

`Framework/CurveFitting/`

and replaced `std::vector` when possible, I also removed some unnecessary includes and unused variables (see for instance `Framework/CurveFitting/test/Functions/ThermalNeutronBk2BkExpAlphaTest.h` and `Framework/CurveFitting/test/Functions/ThermalNeutronBk2BkExpBetaTest.h`). I also used C++11 initialization syntax to remove long sequences of `push_back` (see for instance `Framework/CurveFitting/test/Algorithms/LeBailFunctionTest.h`).

This is part of the work related to #15291 

Release notes: maintenance task, does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
